// @datadog/datadog-api-client/dist/packages/datadog-api-client-v1/types/PromiseAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromiseUsersApi = exports.PromiseUsageMeteringApi = exports.PromiseTagsApi = exports.PromiseSyntheticsApi = exports.PromiseSnapshotsApi = exports.PromiseSlackIntegrationApi = exports.PromiseServiceLevelObjectivesApi = exports.PromiseServiceLevelObjectiveCorrectionsApi = exports.PromiseServiceChecksApi = exports.PromisePagerDutyIntegrationApi = exports.PromiseOrganizationsApi = exports.PromiseNotebooksApi = exports.PromiseMonitorsApi = exports.PromiseMetricsApi = exports.PromiseLogsPipelinesApi = exports.PromiseLogsIndexesApi = exports.PromiseLogsApi = exports.PromiseKeyManagementApi = exports.PromiseIPRangesApi = exports.PromiseHostsApi = exports.PromiseGCPIntegrationApi = exports.PromiseEventsApi = exports.PromiseDowntimesApi = exports.PromiseDashboardsApi = exports.PromiseDashboardListsApi = exports.PromiseAzureIntegrationApi = exports.PromiseAuthenticationApi = exports.PromiseAWSLogsIntegrationApi = exports.PromiseAWSIntegrationApi = void 0;
var ObservableAPI_1 = require("./ObservableAPI");
var PromiseAWSIntegrationApi = /** @class */ (function () {
    function PromiseAWSIntegrationApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableAWSIntegrationApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Create a Datadog-Amazon Web Services integration. Using the `POST` method updates your integration configuration by adding your new configuration to the existing one in your Datadog organization. A unique AWS Account ID for role based authentication.
     * Create an AWS integration
     * @param body AWS Request Object
     */
    PromiseAWSIntegrationApi.prototype.createAWSAccount = function (body, options) {
        var result = this.api.createAWSAccount(body, options);
        return result.toPromise();
    };
    /**
     * Set an AWS tag filter.
     * Set an AWS tag filter
     * @param body Set an AWS tag filter using an &#x60;aws_account_identifier&#x60;, &#x60;namespace&#x60;, and filtering string. Namespace options are &#x60;application_elb&#x60;, &#x60;elb&#x60;, &#x60;lambda&#x60;, &#x60;network_elb&#x60;, &#x60;rds&#x60;, &#x60;sqs&#x60;, and &#x60;custom&#x60;.
     */
    PromiseAWSIntegrationApi.prototype.createAWSTagFilter = function (body, options) {
        var result = this.api.createAWSTagFilter(body, options);
        return result.toPromise();
    };
    /**
     * Generate a new AWS external ID for a given AWS account ID and role name pair.
     * Generate a new external ID
     * @param body Your Datadog role delegation name. For more information about your AWS account Role name, see the [Datadog AWS integration configuration info](https://github.com/DataDog/documentation/blob/master/integrations/amazon_web_services/#installation).
     */
    PromiseAWSIntegrationApi.prototype.createNewAWSExternalID = function (body, options) {
        var result = this.api.createNewAWSExternalID(body, options);
        return result.toPromise();
    };
    /**
     * Delete a Datadog-AWS integration matching the specified `account_id` and `role_name parameters`.
     * Delete an AWS integration
     * @param body AWS request object
     */
    PromiseAWSIntegrationApi.prototype.deleteAWSAccount = function (body, options) {
        var result = this.api.deleteAWSAccount(body, options);
        return result.toPromise();
    };
    /**
     * Delete a tag filtering entry.
     * Delete a tag filtering entry
     * @param body Delete a tag filtering entry for a given AWS account and &#x60;dd-aws&#x60; namespace.
     */
    PromiseAWSIntegrationApi.prototype.deleteAWSTagFilter = function (body, options) {
        var result = this.api.deleteAWSTagFilter(body, options);
        return result.toPromise();
    };
    /**
     * List all Datadog-AWS integrations available in your Datadog organization.
     * List all AWS integrations
     * @param accountId Only return AWS accounts that matches this &#x60;account_id&#x60;.
     * @param roleName Only return AWS accounts that matches this role_name.
     * @param accessKeyId Only return AWS accounts that matches this &#x60;access_key_id&#x60;.
     */
    PromiseAWSIntegrationApi.prototype.listAWSAccounts = function (accountId, roleName, accessKeyId, options) {
        var result = this.api.listAWSAccounts(accountId, roleName, accessKeyId, options);
        return result.toPromise();
    };
    /**
     * Get all AWS tag filters.
     * Get all AWS tag filters
     * @param accountId Only return AWS filters that matches this &#x60;account_id&#x60;.
     */
    PromiseAWSIntegrationApi.prototype.listAWSTagFilters = function (accountId, options) {
        var result = this.api.listAWSTagFilters(accountId, options);
        return result.toPromise();
    };
    /**
     * List all namespace rules for a given Datadog-AWS integration. This endpoint takes no arguments.
     * List namespace rules
     */
    PromiseAWSIntegrationApi.prototype.listAvailableAWSNamespaces = function (options) {
        var result = this.api.listAvailableAWSNamespaces(options);
        return result.toPromise();
    };
    /**
     * Update a Datadog-Amazon Web Services integration.
     * Update an AWS integration
     * @param body AWS request object
     * @param accountId Only return AWS accounts that matches this &#x60;account_id&#x60;.
     * @param roleName Only return AWS accounts that match this &#x60;role_name&#x60;. Required if &#x60;account_id&#x60; is specified.
     * @param accessKeyId Only return AWS accounts that matches this &#x60;access_key_id&#x60;. Required if none of the other two options are specified.
     */
    PromiseAWSIntegrationApi.prototype.updateAWSAccount = function (body, accountId, roleName, accessKeyId, options) {
        var result = this.api.updateAWSAccount(body, accountId, roleName, accessKeyId, options);
        return result.toPromise();
    };
    return PromiseAWSIntegrationApi;
}());
exports.PromiseAWSIntegrationApi = PromiseAWSIntegrationApi;
var ObservableAPI_2 = require("./ObservableAPI");
var PromiseAWSLogsIntegrationApi = /** @class */ (function () {
    function PromiseAWSLogsIntegrationApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_2.ObservableAWSLogsIntegrationApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Test if permissions are present to add a log-forwarding triggers for the given services and AWS account. The input is the same as for Enable an AWS service log collection. Subsequent requests will always repeat the above, so this endpoint can be polled intermittently instead of blocking.  - Returns a status of 'created' when it's checking if the Lambda exists in the account. - Returns a status of 'waiting' while checking. - Returns a status of 'checked and ok' if the Lambda exists. - Returns a status of 'error' if the Lambda does not exist.
     * Check that an AWS Lambda Function exists
     * @param body Check AWS Log Lambda Async request body.
     */
    PromiseAWSLogsIntegrationApi.prototype.checkAWSLogsLambdaAsync = function (body, options) {
        var result = this.api.checkAWSLogsLambdaAsync(body, options);
        return result.toPromise();
    };
    /**
     * Test if permissions are present to add log-forwarding triggers for the given services and AWS account. Input is the same as for `EnableAWSLogServices`. Done async, so can be repeatedly polled in a non-blocking fashion until the async request completes.  - Returns a status of `created` when it's checking if the permissions exists   in the AWS account. - Returns a status of `waiting` while checking. - Returns a status of `checked and ok` if the Lambda exists. - Returns a status of `error` if the Lambda does not exist.
     * Check permissions for log services
     * @param body Check AWS Logs Async Services request body.
     */
    PromiseAWSLogsIntegrationApi.prototype.checkAWSLogsServicesAsync = function (body, options) {
        var result = this.api.checkAWSLogsServicesAsync(body, options);
        return result.toPromise();
    };
    /**
     * Attach the Lambda ARN of the Lambda created for the Datadog-AWS log collection to your AWS account ID to enable log collection.
     * Add AWS Log Lambda ARN
     * @param body AWS Log Lambda Async request body.
     */
    PromiseAWSLogsIntegrationApi.prototype.createAWSLambdaARN = function (body, options) {
        var result = this.api.createAWSLambdaARN(body, options);
        return result.toPromise();
    };
    /**
     * Delete a Datadog-AWS logs configuration by removing the specific Lambda ARN associated with a given AWS account.
     * Delete an AWS Logs integration
     * @param body Delete AWS Lambda ARN request body.
     */
    PromiseAWSLogsIntegrationApi.prototype.deleteAWSLambdaARN = function (body, options) {
        var result = this.api.deleteAWSLambdaARN(body, options);
        return result.toPromise();
    };
    /**
     * Enable automatic log collection for a list of services. This should be run after running `CreateAWSLambdaARN` to save the configuration.
     * Enable an AWS Logs integration
     * @param body Enable AWS Log Services request body.
     */
    PromiseAWSLogsIntegrationApi.prototype.enableAWSLogServices = function (body, options) {
        var result = this.api.enableAWSLogServices(body, options);
        return result.toPromise();
    };
    /**
     * List all Datadog-AWS Logs integrations configured in your Datadog account.
     * List all AWS Logs integrations
     */
    PromiseAWSLogsIntegrationApi.prototype.listAWSLogsIntegrations = function (options) {
        var result = this.api.listAWSLogsIntegrations(options);
        return result.toPromise();
    };
    /**
     * Get the list of current AWS services that Datadog offers automatic log collection. Use returned service IDs with the services parameter for the Enable an AWS service log collection API endpoint.
     * Get list of AWS log ready services
     */
    PromiseAWSLogsIntegrationApi.prototype.listAWSLogsServices = function (options) {
        var result = this.api.listAWSLogsServices(options);
        return result.toPromise();
    };
    return PromiseAWSLogsIntegrationApi;
}());
exports.PromiseAWSLogsIntegrationApi = PromiseAWSLogsIntegrationApi;
var ObservableAPI_3 = require("./ObservableAPI");
var PromiseAuthenticationApi = /** @class */ (function () {
    function PromiseAuthenticationApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_3.ObservableAuthenticationApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Check if the API key (not the APP key) is valid. If invalid, a 403 is returned.
     * Validate API key
     */
    PromiseAuthenticationApi.prototype.validate = function (options) {
        var result = this.api.validate(options);
        return result.toPromise();
    };
    return PromiseAuthenticationApi;
}());
exports.PromiseAuthenticationApi = PromiseAuthenticationApi;
var ObservableAPI_4 = require("./ObservableAPI");
var PromiseAzureIntegrationApi = /** @class */ (function () {
    function PromiseAzureIntegrationApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_4.ObservableAzureIntegrationApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Create a Datadog-Azure integration.  Using the `POST` method updates your integration configuration by adding your new configuration to the existing one in your Datadog organization.  Using the `PUT` method updates your integration configuration by replacing your current configuration with the new one sent to your Datadog organization.
     * Create an Azure integration
     * @param body Create a Datadog-Azure integration for your Datadog account request body.
     */
    PromiseAzureIntegrationApi.prototype.createAzureIntegration = function (body, options) {
        var result = this.api.createAzureIntegration(body, options);
        return result.toPromise();
    };
    /**
     * Delete a given Datadog-Azure integration from your Datadog account.
     * Delete an Azure integration
     * @param body Delete a given Datadog-Azure integration request body.
     */
    PromiseAzureIntegrationApi.prototype.deleteAzureIntegration = function (body, options) {
        var result = this.api.deleteAzureIntegration(body, options);
        return result.toPromise();
    };
    /**
     * List all Datadog-Azure integrations configured in your Datadog account.
     * List all Azure integrations
     */
    PromiseAzureIntegrationApi.prototype.listAzureIntegration = function (options) {
        var result = this.api.listAzureIntegration(options);
        return result.toPromise();
    };
    /**
     * Update the defined list of host filters for a given Datadog-Azure integration.
     * Update Azure integration host filters
     * @param body Update a Datadog-Azure integration&#39;s host filters request body.
     */
    PromiseAzureIntegrationApi.prototype.updateAzureHostFilters = function (body, options) {
        var result = this.api.updateAzureHostFilters(body, options);
        return result.toPromise();
    };
    /**
     * Update a Datadog-Azure integration. Requires an existing `tenant_name` and `client_id`. Any other fields supplied will overwrite existing values. To overwrite `tenant_name` or `client_id`, use `new_tenant_name` and `new_client_id`. To leave a field unchanged, do not supply that field in the payload.
     * Update an Azure integration
     * @param body Update a Datadog-Azure integration request body.
     */
    PromiseAzureIntegrationApi.prototype.updateAzureIntegration = function (body, options) {
        var result = this.api.updateAzureIntegration(body, options);
        return result.toPromise();
    };
    return PromiseAzureIntegrationApi;
}());
exports.PromiseAzureIntegrationApi = PromiseAzureIntegrationApi;
var ObservableAPI_5 = require("./ObservableAPI");
var PromiseDashboardListsApi = /** @class */ (function () {
    function PromiseDashboardListsApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_5.ObservableDashboardListsApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Create an empty dashboard list.
     * Create a dashboard list
     * @param body Create a dashboard list request body.
     */
    PromiseDashboardListsApi.prototype.createDashboardList = function (body, options) {
        var result = this.api.createDashboardList(body, options);
        return result.toPromise();
    };
    /**
     * Delete a dashboard list.
     * Delete a dashboard list
     * @param listId ID of the dashboard list to delete.
     */
    PromiseDashboardListsApi.prototype.deleteDashboardList = function (listId, options) {
        var result = this.api.deleteDashboardList(listId, options);
        return result.toPromise();
    };
    /**
     * Fetch an existing dashboard list's definition.
     * Get a dashboard list
     * @param listId ID of the dashboard list to fetch.
     */
    PromiseDashboardListsApi.prototype.getDashboardList = function (listId, options) {
        var result = this.api.getDashboardList(listId, options);
        return result.toPromise();
    };
    /**
     * Fetch all of your existing dashboard list definitions.
     * Get all dashboard lists
     */
    PromiseDashboardListsApi.prototype.listDashboardLists = function (options) {
        var result = this.api.listDashboardLists(options);
        return result.toPromise();
    };
    /**
     * Update the name of a dashboard list.
     * Update a dashboard list
     * @param listId ID of the dashboard list to update.
     * @param body Update a dashboard list request body.
     */
    PromiseDashboardListsApi.prototype.updateDashboardList = function (listId, body, options) {
        var result = this.api.updateDashboardList(listId, body, options);
        return result.toPromise();
    };
    return PromiseDashboardListsApi;
}());
exports.PromiseDashboardListsApi = PromiseDashboardListsApi;
var ObservableAPI_6 = require("./ObservableAPI");
var PromiseDashboardsApi = /** @class */ (function () {
    function PromiseDashboardsApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_6.ObservableDashboardsApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Create a dashboard using the specified options. When defining queries in your widgets, take note of which queries should have the `as_count()` or `as_rate()` modifiers appended. Refer to the following [documentation](https://docs.datadoghq.com/developers/metrics/type_modifiers/?tab=count#in-application-modifiers) for more information on these modifiers.
     * Create a new dashboard
     * @param body Create a dashboard request body.
     */
    PromiseDashboardsApi.prototype.createDashboard = function (body, options) {
        var result = this.api.createDashboard(body, options);
        return result.toPromise();
    };
    /**
     * Delete a dashboard using the specified ID.
     * Delete a dashboard
     * @param dashboardId The ID of the dashboard.
     */
    PromiseDashboardsApi.prototype.deleteDashboard = function (dashboardId, options) {
        var result = this.api.deleteDashboard(dashboardId, options);
        return result.toPromise();
    };
    /**
     * Get a dashboard using the specified ID.
     * Get a dashboard
     * @param dashboardId The ID of the dashboard.
     */
    PromiseDashboardsApi.prototype.getDashboard = function (dashboardId, options) {
        var result = this.api.getDashboard(dashboardId, options);
        return result.toPromise();
    };
    /**
     * Get all dashboards.  **Note**: This query will only return custom created or cloned dashboards. This query will not return preset dashboards.
     * Get all dashboards
     * @param filterShared When &#x60;true&#x60;, this query only returns shared custom created or cloned dashboards.
     */
    PromiseDashboardsApi.prototype.listDashboards = function (filterShared, options) {
        var result = this.api.listDashboards(filterShared, options);
        return result.toPromise();
    };
    /**
     * Update a dashboard using the specified ID.
     * Update a dashboard
     * @param dashboardId The ID of the dashboard.
     * @param body Update Dashboard request body.
     */
    PromiseDashboardsApi.prototype.updateDashboard = function (dashboardId, body, options) {
        var result = this.api.updateDashboard(dashboardId, body, options);
        return result.toPromise();
    };
    return PromiseDashboardsApi;
}());
exports.PromiseDashboardsApi = PromiseDashboardsApi;
var ObservableAPI_7 = require("./ObservableAPI");
var PromiseDowntimesApi = /** @class */ (function () {
    function PromiseDowntimesApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_7.ObservableDowntimesApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Cancel a downtime.
     * Cancel a downtime
     * @param downtimeId ID of the downtime to cancel.
     */
    PromiseDowntimesApi.prototype.cancelDowntime = function (downtimeId, options) {
        var result = this.api.cancelDowntime(downtimeId, options);
        return result.toPromise();
    };
    /**
     * Delete all downtimes that match the scope of `X`.
     * Cancel downtimes by scope
     * @param body Scope to cancel downtimes for.
     */
    PromiseDowntimesApi.prototype.cancelDowntimesByScope = function (body, options) {
        var result = this.api.cancelDowntimesByScope(body, options);
        return result.toPromise();
    };
    /**
     * Schedule a downtime.
     * Schedule a downtime
     * @param body Schedule a downtime request body.
     */
    PromiseDowntimesApi.prototype.createDowntime = function (body, options) {
        var result = this.api.createDowntime(body, options);
        return result.toPromise();
    };
    /**
     * Get downtime detail by `downtime_id`.
     * Get a downtime
     * @param downtimeId ID of the downtime to fetch.
     */
    PromiseDowntimesApi.prototype.getDowntime = function (downtimeId, options) {
        var result = this.api.getDowntime(downtimeId, options);
        return result.toPromise();
    };
    /**
     * Get all scheduled downtimes.
     * Get all downtimes
     * @param currentOnly Only return downtimes that are active when the request is made.
     */
    PromiseDowntimesApi.prototype.listDowntimes = function (currentOnly, options) {
        var result = this.api.listDowntimes(currentOnly, options);
        return result.toPromise();
    };
    /**
     * Get all downtimes for the specified monitor
     * Get all downtimes for a monitor
     * @param monitorId The id of the monitor
     */
    PromiseDowntimesApi.prototype.listMonitorDowntimes = function (monitorId, options) {
        var result = this.api.listMonitorDowntimes(monitorId, options);
        return result.toPromise();
    };
    /**
     * Update a single downtime by `downtime_id`.
     * Update a downtime
     * @param downtimeId ID of the downtime to update.
     * @param body Update a downtime request body.
     */
    PromiseDowntimesApi.prototype.updateDowntime = function (downtimeId, body, options) {
        var result = this.api.updateDowntime(downtimeId, body, options);
        return result.toPromise();
    };
    return PromiseDowntimesApi;
}());
exports.PromiseDowntimesApi = PromiseDowntimesApi;
var ObservableAPI_8 = require("./ObservableAPI");
var PromiseEventsApi = /** @class */ (function () {
    function PromiseEventsApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_8.ObservableEventsApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * This endpoint allows you to post events to the stream. Tag them, set priority and event aggregate them with other events.
     * Post an event
     * @param body Event request object
     */
    PromiseEventsApi.prototype.createEvent = function (body, options) {
        var result = this.api.createEvent(body, options);
        return result.toPromise();
    };
    /**
     * This endpoint allows you to query for event details.  **Note**: If the event you’re querying contains markdown formatting of any kind, you may see characters such as `%`,`\\`,`n` in your output.
     * Get an event
     * @param eventId The ID of the event.
     */
    PromiseEventsApi.prototype.getEvent = function (eventId, options) {
        var result = this.api.getEvent(eventId, options);
        return result.toPromise();
    };
    /**
     * The event stream can be queried and filtered by time, priority, sources and tags.  **Notes**: - If the event you’re querying contains markdown formatting of any kind, you may see characters such as `%`,`\\`,`n` in your output.  - This endpoint returns a maximum of `1000` most recent results. To return additional results, identify the last timestamp of the last result and set that as the `end` query time to paginate the results. You can also use the page parameter to specify which set of `1000` results to return.
     * Query the event stream
     * @param start POSIX timestamp.
     * @param end POSIX timestamp.
     * @param priority Priority of your events, either &#x60;low&#x60; or &#x60;normal&#x60;.
     * @param sources A comma separated string of sources.
     * @param tags A comma separated list indicating what tags, if any, should be used to filter the list of monitors by scope.
     * @param unaggregated Set unaggregated to &#x60;true&#x60; to return all events within the specified [&#x60;start&#x60;,&#x60;end&#x60;] timeframe. Otherwise if an event is aggregated to a parent event with a timestamp outside of the timeframe, it won&#39;t be available in the output. Aggregated events with &#x60;is_aggregate&#x3D;true&#x60; in the response will still be returned unless exclude_aggregate is set to &#x60;true.&#x60;
     * @param excludeAggregate Set &#x60;exclude_aggregate&#x60; to &#x60;true&#x60; to only return unaggregated events where &#x60;is_aggregate&#x3D;false&#x60; in the response. If the &#x60;exclude_aggregate&#x60; parameter is set to &#x60;true&#x60;, then the unaggregated parameter is ignored and will be &#x60;true&#x60; by default.
     * @param page By default 1000 results are returned per request. Set page to the number of the page to return with &#x60;0&#x60; being the first page. The page parameter can only be used when either unaggregated or exclude_aggregate is set to &#x60;true.&#x60;
     */
    PromiseEventsApi.prototype.listEvents = function (start, end, priority, sources, tags, unaggregated, excludeAggregate, page, options) {
        var result = this.api.listEvents(start, end, priority, sources, tags, unaggregated, excludeAggregate, page, options);
        return result.toPromise();
    };
    return PromiseEventsApi;
}());
exports.PromiseEventsApi = PromiseEventsApi;
var ObservableAPI_9 = require("./ObservableAPI");
var PromiseGCPIntegrationApi = /** @class */ (function () {
    function PromiseGCPIntegrationApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_9.ObservableGCPIntegrationApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Create a Datadog-GCP integration.
     * Create a GCP integration
     * @param body Create a Datadog-GCP integration.
     */
    PromiseGCPIntegrationApi.prototype.createGCPIntegration = function (body, options) {
        var result = this.api.createGCPIntegration(body, options);
        return result.toPromise();
    };
    /**
     * Delete a given Datadog-GCP integration.
     * Delete a GCP integration
     * @param body Delete a given Datadog-GCP integration.
     */
    PromiseGCPIntegrationApi.prototype.deleteGCPIntegration = function (body, options) {
        var result = this.api.deleteGCPIntegration(body, options);
        return result.toPromise();
    };
    /**
     * List all Datadog-GCP integrations configured in your Datadog account.
     * List all GCP integrations
     */
    PromiseGCPIntegrationApi.prototype.listGCPIntegration = function (options) {
        var result = this.api.listGCPIntegration(options);
        return result.toPromise();
    };
    /**
     * Update a Datadog-GCP integrations host_filters and/or auto-mute. Requires a `project_id` and `client_email`, however these fields cannot be updated. If you need to update these fields, delete and use the create (`POST`) endpoint. The unspecified fields will keep their original values.
     * Update a GCP integration
     * @param body Update a Datadog-GCP integration.
     */
    PromiseGCPIntegrationApi.prototype.updateGCPIntegration = function (body, options) {
        var result = this.api.updateGCPIntegration(body, options);
        return result.toPromise();
    };
    return PromiseGCPIntegrationApi;
}());
exports.PromiseGCPIntegrationApi = PromiseGCPIntegrationApi;
var ObservableAPI_10 = require("./ObservableAPI");
var PromiseHostsApi = /** @class */ (function () {
    function PromiseHostsApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_10.ObservableHostsApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * This endpoint returns the total number of active and up hosts in your Datadog account. Active means the host has reported in the past hour, and up means it has reported in the past two hours.
     * Get the total number of active hosts
     * @param from Number of seconds from which you want to get total number of active hosts.
     */
    PromiseHostsApi.prototype.getHostTotals = function (from, options) {
        var result = this.api.getHostTotals(from, options);
        return result.toPromise();
    };
    /**
     * This endpoint allows searching for hosts by name, alias, or tag. Hosts live within the past 3 hours are included by default. Retention is 7 days. Results are paginated with a max of 1000 results at a time.
     * Get all hosts for your organization
     * @param filter String to filter search results.
     * @param sortField Sort hosts by this field.
     * @param sortDir Direction of sort. Options include &#x60;asc&#x60; and &#x60;desc&#x60;.
     * @param start Host result to start search from.
     * @param count Number of hosts to return. Max 1000.
     * @param from Number of seconds since UNIX epoch from which you want to search your hosts.
     * @param includeMutedHostsData Include information on the muted status of hosts and when the mute expires.
     * @param includeHostsMetadata Include additional metadata about the hosts (agent_version, machine, platform, processor, etc.).
     */
    PromiseHostsApi.prototype.listHosts = function (filter, sortField, sortDir, start, count, from, includeMutedHostsData, includeHostsMetadata, options) {
        var result = this.api.listHosts(filter, sortField, sortDir, start, count, from, includeMutedHostsData, includeHostsMetadata, options);
        return result.toPromise();
    };
    /**
     * Mute a host.
     * Mute a host
     * @param hostName Name of the host to mute.
     * @param body Mute a host request body.
     */
    PromiseHostsApi.prototype.muteHost = function (hostName, body, options) {
        var result = this.api.muteHost(hostName, body, options);
        return result.toPromise();
    };
    /**
     * Unmutes a host. This endpoint takes no JSON arguments.
     * Unmute a host
     * @param hostName Name of the host to unmute.
     */
    PromiseHostsApi.prototype.unmuteHost = function (hostName, options) {
        var result = this.api.unmuteHost(hostName, options);
        return result.toPromise();
    };
    return PromiseHostsApi;
}());
exports.PromiseHostsApi = PromiseHostsApi;
var ObservableAPI_11 = require("./ObservableAPI");
var PromiseIPRangesApi = /** @class */ (function () {
    function PromiseIPRangesApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_11.ObservableIPRangesApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Get information about Datadog IP ranges.
     * List IP Ranges
     */
    PromiseIPRangesApi.prototype.getIPRanges = function (options) {
        var result = this.api.getIPRanges(options);
        return result.toPromise();
    };
    return PromiseIPRangesApi;
}());
exports.PromiseIPRangesApi = PromiseIPRangesApi;
var ObservableAPI_12 = require("./ObservableAPI");
var PromiseKeyManagementApi = /** @class */ (function () {
    function PromiseKeyManagementApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_12.ObservableKeyManagementApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Creates an API key with a given name.
     * Create an API key
     * @param body
     */
    PromiseKeyManagementApi.prototype.createAPIKey = function (body, options) {
        var result = this.api.createAPIKey(body, options);
        return result.toPromise();
    };
    /**
     * Create an application key with a given name.
     * Create an application key
     * @param body
     */
    PromiseKeyManagementApi.prototype.createApplicationKey = function (body, options) {
        var result = this.api.createApplicationKey(body, options);
        return result.toPromise();
    };
    /**
     * Delete a given API key.
     * Delete an API key
     * @param key The specific API key you are working with.
     */
    PromiseKeyManagementApi.prototype.deleteAPIKey = function (key, options) {
        var result = this.api.deleteAPIKey(key, options);
        return result.toPromise();
    };
    /**
     * Delete a given application key.
     * Delete an application key
     * @param key The specific APP key you are working with.
     */
    PromiseKeyManagementApi.prototype.deleteApplicationKey = function (key, options) {
        var result = this.api.deleteApplicationKey(key, options);
        return result.toPromise();
    };
    /**
     * Get a given API key.
     * Get API key
     * @param key The specific API key you are working with.
     */
    PromiseKeyManagementApi.prototype.getAPIKey = function (key, options) {
        var result = this.api.getAPIKey(key, options);
        return result.toPromise();
    };
    /**
     * Get a given application key.
     * Get an application key
     * @param key The specific APP key you are working with.
     */
    PromiseKeyManagementApi.prototype.getApplicationKey = function (key, options) {
        var result = this.api.getApplicationKey(key, options);
        return result.toPromise();
    };
    /**
     * Get all API keys available for your account.
     * Get all API keys
     */
    PromiseKeyManagementApi.prototype.listAPIKeys = function (options) {
        var result = this.api.listAPIKeys(options);
        return result.toPromise();
    };
    /**
     * Get all application keys available for your Datadog account.
     * Get all application keys
     */
    PromiseKeyManagementApi.prototype.listApplicationKeys = function (options) {
        var result = this.api.listApplicationKeys(options);
        return result.toPromise();
    };
    /**
     * Edit an API key name.
     * Edit an API key
     * @param key The specific API key you are working with.
     * @param body
     */
    PromiseKeyManagementApi.prototype.updateAPIKey = function (key, body, options) {
        var result = this.api.updateAPIKey(key, body, options);
        return result.toPromise();
    };
    /**
     * Edit an application key name.
     * Edit an application key
     * @param key The specific APP key you are working with.
     * @param body
     */
    PromiseKeyManagementApi.prototype.updateApplicationKey = function (key, body, options) {
        var result = this.api.updateApplicationKey(key, body, options);
        return result.toPromise();
    };
    return PromiseKeyManagementApi;
}());
exports.PromiseKeyManagementApi = PromiseKeyManagementApi;
var ObservableAPI_13 = require("./ObservableAPI");
var PromiseLogsApi = /** @class */ (function () {
    function PromiseLogsApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_13.ObservableLogsApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * List endpoint returns logs that match a log search query. [Results are paginated][1].  **If you are considering archiving logs for your organization, consider use of the Datadog archive capabilities instead of the log list API. See [Datadog Logs Archive documentation][2].**  [1]: /logs/guide/collect-multiple-logs-with-pagination [2]: https://docs.datadoghq.com/logs/archives
     * Search logs
     * @param body Logs filter
     */
    PromiseLogsApi.prototype.listLogs = function (body, options) {
        var result = this.api.listLogs(body, options);
        return result.toPromise();
    };
    /**
     * Send your logs to your Datadog platform over HTTP. Limits per HTTP request are:  - Maximum content size per payload (uncompressed): 5MB - Maximum size for a single log: 1MB - Maximum array size if sending multiple logs in an array: 1000 entries  Any log exceeding 1MB is accepted and truncated by Datadog: - For a single log request, the API truncates the log at 1MB and returns a 2xx. - For a multi-logs request, the API processes all logs, truncates only logs larger than 1MB, and returns a 2xx.  Datadog recommends sending your logs compressed. Add the `Content-Encoding: gzip` header to the request when sending compressed logs.  The status codes answered by the HTTP API are: - 200: OK - 400: Bad request (likely an issue in the payload formatting) - 403: Permission issue (likely using an invalid API Key) - 413: Payload too large (batch is above 5MB uncompressed) - 5xx: Internal error, request should be retried after some time
     * Send logs
     * @param body Log to send (JSON format).
     * @param contentEncoding HTTP header used to compress the media-type.
     * @param ddtags Log tags can be passed as query parameters with &#x60;text/plain&#x60; content type.
     */
    PromiseLogsApi.prototype.submitLog = function (body, contentEncoding, ddtags, options) {
        var result = this.api.submitLog(body, contentEncoding, ddtags, options);
        return result.toPromise();
    };
    return PromiseLogsApi;
}());
exports.PromiseLogsApi = PromiseLogsApi;
var ObservableAPI_14 = require("./ObservableAPI");
var PromiseLogsIndexesApi = /** @class */ (function () {
    function PromiseLogsIndexesApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_14.ObservableLogsIndexesApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Creates a new index. Returns the Index object passed in the request body when the request is successful.
     * Create an index
     * @param body Object containing the new index.
     */
    PromiseLogsIndexesApi.prototype.createLogsIndex = function (body, options) {
        var result = this.api.createLogsIndex(body, options);
        return result.toPromise();
    };
    /**
     * Get one log index from your organization. This endpoint takes no JSON arguments.
     * Get an index
     * @param name Name of the log index.
     */
    PromiseLogsIndexesApi.prototype.getLogsIndex = function (name, options) {
        var result = this.api.getLogsIndex(name, options);
        return result.toPromise();
    };
    /**
     * Get the current order of your log indexes. This endpoint takes no JSON arguments.
     * Get indexes order
     */
    PromiseLogsIndexesApi.prototype.getLogsIndexOrder = function (options) {
        var result = this.api.getLogsIndexOrder(options);
        return result.toPromise();
    };
    /**
     * The Index object describes the configuration of a log index. This endpoint returns an array of the `LogIndex` objects of your organization.
     * Get all indexes
     */
    PromiseLogsIndexesApi.prototype.listLogIndexes = function (options) {
        var result = this.api.listLogIndexes(options);
        return result.toPromise();
    };
    /**
     * Update an index as identified by its name. Returns the Index object passed in the request body when the request is successful.  Using the `PUT` method updates your index’s configuration by **replacing** your current configuration with the new one sent to your Datadog organization.
     * Update an index
     * @param name Name of the log index.
     * @param body Object containing the new &#x60;LogsIndexUpdateRequest&#x60;.
     */
    PromiseLogsIndexesApi.prototype.updateLogsIndex = function (name, body, options) {
        var result = this.api.updateLogsIndex(name, body, options);
        return result.toPromise();
    };
    /**
     * This endpoint updates the index order of your organization. It returns the index order object passed in the request body when the request is successful.
     * Update indexes order
     * @param body Object containing the new ordered list of index names
     */
    PromiseLogsIndexesApi.prototype.updateLogsIndexOrder = function (body, options) {
        var result = this.api.updateLogsIndexOrder(body, options);
        return result.toPromise();
    };
    return PromiseLogsIndexesApi;
}());
exports.PromiseLogsIndexesApi = PromiseLogsIndexesApi;
var ObservableAPI_15 = require("./ObservableAPI");
var PromiseLogsPipelinesApi = /** @class */ (function () {
    function PromiseLogsPipelinesApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_15.ObservableLogsPipelinesApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Create a pipeline in your organization.
     * Create a pipeline
     * @param body Definition of the new pipeline.
     */
    PromiseLogsPipelinesApi.prototype.createLogsPipeline = function (body, options) {
        var result = this.api.createLogsPipeline(body, options);
        return result.toPromise();
    };
    /**
     * Delete a given pipeline from your organization. This endpoint takes no JSON arguments.
     * Delete a pipeline
     * @param pipelineId ID of the pipeline to delete.
     */
    PromiseLogsPipelinesApi.prototype.deleteLogsPipeline = function (pipelineId, options) {
        var result = this.api.deleteLogsPipeline(pipelineId, options);
        return result.toPromise();
    };
    /**
     * Get a specific pipeline from your organization. This endpoint takes no JSON arguments.
     * Get a pipeline
     * @param pipelineId ID of the pipeline to get.
     */
    PromiseLogsPipelinesApi.prototype.getLogsPipeline = function (pipelineId, options) {
        var result = this.api.getLogsPipeline(pipelineId, options);
        return result.toPromise();
    };
    /**
     * Get the current order of your pipelines. This endpoint takes no JSON arguments.
     * Get pipeline order
     */
    PromiseLogsPipelinesApi.prototype.getLogsPipelineOrder = function (options) {
        var result = this.api.getLogsPipelineOrder(options);
        return result.toPromise();
    };
    /**
     * Get all pipelines from your organization. This endpoint takes no JSON arguments.
     * Get all pipelines
     */
    PromiseLogsPipelinesApi.prototype.listLogsPipelines = function (options) {
        var result = this.api.listLogsPipelines(options);
        return result.toPromise();
    };
    /**
     * Update a given pipeline configuration to change it’s processors or their order.  **Note**: Using this method updates your pipeline configuration by **replacing** your current configuration with the new one sent to your Datadog organization.
     * Update a pipeline
     * @param pipelineId ID of the pipeline to delete.
     * @param body New definition of the pipeline.
     */
    PromiseLogsPipelinesApi.prototype.updateLogsPipeline = function (pipelineId, body, options) {
        var result = this.api.updateLogsPipeline(pipelineId, body, options);
        return result.toPromise();
    };
    /**
     * Update the order of your pipelines. Since logs are processed sequentially, reordering a pipeline may change the structure and content of the data processed by other pipelines and their processors.  **Note**: Using the `PUT` method updates your pipeline order by replacing your current order with the new one sent to your Datadog organization.
     * Update pipeline order
     * @param body Object containing the new ordered list of pipeline IDs.
     */
    PromiseLogsPipelinesApi.prototype.updateLogsPipelineOrder = function (body, options) {
        var result = this.api.updateLogsPipelineOrder(body, options);
        return result.toPromise();
    };
    return PromiseLogsPipelinesApi;
}());
exports.PromiseLogsPipelinesApi = PromiseLogsPipelinesApi;
var ObservableAPI_16 = require("./ObservableAPI");
var PromiseMetricsApi = /** @class */ (function () {
    function PromiseMetricsApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_16.ObservableMetricsApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Get metadata about a specific metric.
     * Get metric metadata
     * @param metricName Name of the metric for which to get metadata.
     */
    PromiseMetricsApi.prototype.getMetricMetadata = function (metricName, options) {
        var result = this.api.getMetricMetadata(metricName, options);
        return result.toPromise();
    };
    /**
     * Get the list of actively reporting metrics from a given time until now.
     * Get active metrics list
     * @param from Seconds since the Unix epoch.
     * @param host Hostname for filtering the list of metrics returned. If set, metrics retrieved are those with the corresponding hostname tag.
     * @param tagFilter Filter metrics that have been submitted with the given tags. Supports boolean and wildcard expressions. Cannot be combined with other filters.
     */
    PromiseMetricsApi.prototype.listActiveMetrics = function (from, host, tagFilter, options) {
        var result = this.api.listActiveMetrics(from, host, tagFilter, options);
        return result.toPromise();
    };
    /**
     * Search for metrics from the last 24 hours in Datadog.
     * Search metrics
     * @param q Query string to search metrics upon. Must be prefixed with &#x60;metrics:&#x60;.
     */
    PromiseMetricsApi.prototype.listMetrics = function (q, options) {
        var result = this.api.listMetrics(q, options);
        return result.toPromise();
    };
    /**
     * Query timeseries points.
     * Query timeseries points
     * @param from Start of the queried time period, seconds since the Unix epoch.
     * @param to End of the queried time period, seconds since the Unix epoch.
     * @param query Query string.
     */
    PromiseMetricsApi.prototype.queryMetrics = function (from, to, query, options) {
        var result = this.api.queryMetrics(from, to, query, options);
        return result.toPromise();
    };
    /**
     * The metrics end-point allows you to post time-series data that can be graphed on Datadog’s dashboards. The maximum payload size is 3.2 megabytes (3200000 bytes). Compressed payloads must have a decompressed size of less than 62 megabytes (62914560 bytes).  If you’re submitting metrics directly to the Datadog API without using DogStatsD, expect:  - 64 bits for the timestamp - 32 bits for the value - 20 bytes for the metric names - 50 bytes for the timeseries - The full payload is approximately 100 bytes. However, with the DogStatsD API, compression is applied, which reduces the payload size.
     * Submit metrics
     * @param body
     */
    PromiseMetricsApi.prototype.submitMetrics = function (body, options) {
        var result = this.api.submitMetrics(body, options);
        return result.toPromise();
    };
    /**
     * Edit metadata of a specific metric. Find out more about [supported types](https://docs.datadoghq.com/developers/metrics).
     * Edit metric metadata
     * @param metricName Name of the metric for which to edit metadata.
     * @param body New metadata.
     */
    PromiseMetricsApi.prototype.updateMetricMetadata = function (metricName, body, options) {
        var result = this.api.updateMetricMetadata(metricName, body, options);
        return result.toPromise();
    };
    return PromiseMetricsApi;
}());
exports.PromiseMetricsApi = PromiseMetricsApi;
var ObservableAPI_17 = require("./ObservableAPI");
var PromiseMonitorsApi = /** @class */ (function () {
    function PromiseMonitorsApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_17.ObservableMonitorsApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Check if the given monitors can be deleted.
     * Check if a monitor can be deleted
     * @param monitorIds The IDs of the monitor to check.
     */
    PromiseMonitorsApi.prototype.checkCanDeleteMonitor = function (monitorIds, options) {
        var result = this.api.checkCanDeleteMonitor(monitorIds, options);
        return result.toPromise();
    };
    /**
     * Create a monitor using the specified options.  #### Monitor Types  The type of monitor chosen from:  - anomaly: `query alert` - APM: `query alert` or `trace-analytics alert` - composite: `composite` - custom: `service check` - event: `event alert` - forecast: `query alert` - host: `service check` - integration: `query alert` or `service check` - live process: `process alert` - logs: `log alert` - metric: `metric alert` - network: `service check` - outlier: `query alert` - process: `service check` - rum: `rum alert` - SLO: `slo alert` - watchdog: `event alert` - event-v2: `event-v2 alert`  #### Query Types  **Metric Alert Query**  Example: `time_aggr(time_window):space_aggr:metric{tags} [by {key}] operator #`  - `time_aggr`: avg, sum, max, min, change, or pct_change - `time_window`: `last_#m` (with `#` between 1 and 2880 depending on the monitor type) or `last_#h`(with `#` between 1 and 48 depending on the monitor type), or `last_1d` - `space_aggr`: avg, sum, min, or max - `tags`: one or more tags (comma-separated), or * - `key`: a 'key' in key:value tag syntax; defines a separate alert for each tag in the group (multi-alert) - `operator`: <, <=, >, >=, ==, or != - `#`: an integer or decimal number used to set the threshold  If you are using the `_change_` or `_pct_change_` time aggregator, instead use `change_aggr(time_aggr(time_window), timeshift):space_aggr:metric{tags} [by {key}] operator #` with:  - `change_aggr` change, pct_change - `time_aggr` avg, sum, max, min [Learn more](https://docs.datadoghq.com/monitors/monitor_types/#define-the-conditions) - `time_window` last\\_#m (between 1 and 2880 depending on the monitor type), last\\_#h (between 1 and 48 depending on the monitor type), or last_#d (1 or 2) - `timeshift` #m_ago (5, 10, 15, or 30), #h_ago (1, 2, or 4), or 1d_ago  Use this to create an outlier monitor using the following query: `avg(last_30m):outliers(avg:system.cpu.user{role:es-events-data} by {host}, 'dbscan', 7) > 0`  **Service Check Query**  Example: `\"check\".over(tags).last(count).by(group).count_by_status()`  - **`check`** name of the check, e.g. `datadog.agent.up` - **`tags`** one or more quoted tags (comma-separated), or \"*\". e.g.: `.over(\"env:prod\", \"role:db\")`; **`over`** cannot be blank. - **`count`** must be at greater than or equal to your max threshold (defined in the `options`). It is limited to 100. For example, if you've specified to notify on 1 critical, 3 ok, and 2 warn statuses, `count` should be at least 3. - **`group`** must be specified for check monitors. Per-check grouping is already explicitly known for some service checks. For example, Postgres integration monitors are tagged by `db`, `host`, and `port`, and Network monitors by `host`, `instance`, and `url`. See [Service Checks](https://docs.datadoghq.com/api/latest/service-checks/) documentation for more information.  **Event Alert Query**  Example: `events('sources:nagios status:error,warning priority:normal tags: \"string query\"').rollup(\"count\").last(\"1h\")\"`  - **`event`**, the event query string: - **`string_query`** free text query to match against event title and text. - **`sources`** event sources (comma-separated). - **`status`** event statuses (comma-separated). Valid options: error, warn, and info. - **`priority`** event priorities (comma-separated). Valid options: low, normal, all. - **`host`** event reporting host (comma-separated). - **`tags`** event tags (comma-separated). - **`excluded_tags`** excluded event tags (comma-separated). - **`rollup`** the stats roll-up method. `count` is the only supported method now. - **`last`** the timeframe to roll up the counts. Examples: 45m, 4h. Supported timeframes: m, h and d. This value should not exceed 48 hours.  **NOTE** Only available on US1 and EU.  **Event V2 Alert Query**  Example: `events(query).rollup(rollup_method[, measure]).last(time_window) operator #`  - **`query`** The search query - following the [Log search syntax](https://docs.datadoghq.com/logs/search_syntax/). - **`rollup_method`** The stats roll-up method - supports `count`, `avg` and `cardinality`. - **`measure`** For `avg` and cardinality `rollup_method` - specify the measure or the facet name you want to use. - **`time_window`** #m (5, 10, 15, or 30), #h (1, 2, or 4, 24). - **`operator`** `<`, `<=`, `>`, `>=`, `==`, or `!=`. - **`#`** an integer or decimal number used to set the threshold.  **NOTE** Only available on US1-FED, US3, and in closed beta on EU and US1.  **Process Alert Query**  Example: `processes(search).over(tags).rollup('count').last(timeframe) operator #`  - **`search`** free text search string for querying processes. Matching processes match results on the [Live Processes](https://docs.datadoghq.com/infrastructure/process/?tab=linuxwindows) page. - **`tags`** one or more tags (comma-separated) - **`timeframe`** the timeframe to roll up the counts. Examples: 10m, 4h. Supported timeframes: s, m, h and d - **`operator`** <, <=, >, >=, ==, or != - **`#`** an integer or decimal number used to set the threshold  **Logs Alert Query**  Example: `logs(query).index(index_name).rollup(rollup_method[, measure]).last(time_window) operator #`  - **`query`** The search query - following the [Log search syntax](https://docs.datadoghq.com/logs/search_syntax/). - **`index_name`** For multi-index organizations, the log index in which the request is performed. - **`rollup_method`** The stats roll-up method - supports `count`, `avg` and `cardinality`. - **`measure`** For `avg` and cardinality `rollup_method` - specify the measure or the facet name you want to use. - **`time_window`** #m (between 1 and 2880), #h (between 1 and 48) - **`operator`** `<`, `<=`, `>`, `>=`, `==`, or `!=`. - **`#`** an integer or decimal number used to set the threshold.  **Composite Query**  Example: `12345 && 67890`, where `12345` and `67890` are the IDs of non-composite monitors  * **`name`** [*required*, *default* = **dynamic, based on query**]: The name of the alert. * **`message`** [*required*, *default* = **dynamic, based on query**]: A message to include with notifications for this monitor. Email notifications can be sent to specific users by using the same '@username' notation as events. * **`tags`** [*optional*, *default* = **empty list**]: A list of tags to associate with your monitor. When getting all monitor details via the API, use the `monitor_tags` argument to filter results by these tags. It is only available via the API and isn't visible or editable in the Datadog UI.  **SLO Alert Query**  Example: `error_budget(\"slo_id\").over(\"time_window\") operator #`  - **`slo_id`**: The alphanumeric SLO ID of the SLO you are configuring the alert for. - **`time_window`**: The time window of the SLO target you wish to alert on. Valid options: `7d`, `30d`, `90d`. - **`operator`**: `>=` or `>`
     * Create a monitor
     * @param body Create a monitor request body.
     */
    PromiseMonitorsApi.prototype.createMonitor = function (body, options) {
        var result = this.api.createMonitor(body, options);
        return result.toPromise();
    };
    /**
     * Delete the specified monitor
     * Delete a monitor
     * @param monitorId The ID of the monitor.
     * @param force Delete the monitor even if it&#39;s referenced by other resources (e.g. SLO, composite monitor).
     */
    PromiseMonitorsApi.prototype.deleteMonitor = function (monitorId, force, options) {
        var result = this.api.deleteMonitor(monitorId, force, options);
        return result.toPromise();
    };
    /**
     * Get details about the specified monitor from your organization.
     * Get a monitor's details
     * @param monitorId The ID of the monitor
     * @param groupStates When specified, shows additional information about the group states. Choose one or more from &#x60;all&#x60;, &#x60;alert&#x60;, &#x60;warn&#x60;, and &#x60;no data&#x60;.
     */
    PromiseMonitorsApi.prototype.getMonitor = function (monitorId, groupStates, options) {
        var result = this.api.getMonitor(monitorId, groupStates, options);
        return result.toPromise();
    };
    /**
     * Get details about the specified monitor from your organization.
     * Get all monitor details
     * @param groupStates When specified, shows additional information about the group states. Choose one or more from &#x60;all&#x60;, &#x60;alert&#x60;, &#x60;warn&#x60;, and &#x60;no data&#x60;.
     * @param name A string to filter monitors by name.
     * @param tags A comma separated list indicating what tags, if any, should be used to filter the list of monitors by scope. For example, &#x60;host:host0&#x60;.
     * @param monitorTags A comma separated list indicating what service and/or custom tags, if any, should be used to filter the list of monitors. Tags created in the Datadog UI automatically have the service key prepended. For example, &#x60;service:my-app&#x60;.
     * @param withDowntimes If this argument is set to true, then the returned data includes all current downtimes for each monitor.
     * @param idOffset Monitor ID offset.
     * @param page The page to start paginating from. If this argument is not specified, the request returns all monitors without pagination.
     * @param pageSize The number of monitors to return per page. If the page argument is not specified, the default behavior returns all monitors without a &#x60;page_size&#x60; limit. However, if page is specified and &#x60;page_size&#x60; is not, the argument defaults to 100.
     */
    PromiseMonitorsApi.prototype.listMonitors = function (groupStates, name, tags, monitorTags, withDowntimes, idOffset, page, pageSize, options) {
        var result = this.api.listMonitors(groupStates, name, tags, monitorTags, withDowntimes, idOffset, page, pageSize, options);
        return result.toPromise();
    };
    /**
     * Search and filter your monitor groups details.
     * Monitors group search
     * @param query After entering a search query in your [Manage Monitor page][1] use the query parameter value in the URL of the page as value for this parameter. Consult the dedicated [manage monitor documentation][2] page to learn more.  The query can contain any number of space-separated monitor attributes, for instance &#x60;query&#x3D;\&quot;type:metric status:alert\&quot;&#x60;.  [1]: https://app.datadoghq.com/monitors/manage [2]: /monitors/manage_monitor/#find-the-monitors
     * @param page Page to start paginating from.
     * @param perPage Number of monitors to return per page.
     * @param sort String for sort order, composed of field and sort order separate by a comma, e.g. &#x60;name,asc&#x60;. Supported sort directions: &#x60;asc&#x60;, &#x60;desc&#x60;. Supported fields:  * &#x60;name&#x60; * &#x60;status&#x60; * &#x60;tags&#x60;
     */
    PromiseMonitorsApi.prototype.searchMonitorGroups = function (query, page, perPage, sort, options) {
        var result = this.api.searchMonitorGroups(query, page, perPage, sort, options);
        return result.toPromise();
    };
    /**
     * Search and filter your monitors details.
     * Monitors search
     * @param query After entering a search query in your [Manage Monitor page][1] use the query parameter value in the URL of the page as value for this parameter. Consult the dedicated [manage monitor documentation][2] page to learn more.  The query can contain any number of space-separated monitor attributes, for instance &#x60;query&#x3D;\&quot;type:metric status:alert\&quot;&#x60;.  [1]: https://app.datadoghq.com/monitors/manage [2]: /monitors/manage_monitor/#find-the-monitors
     * @param page Page to start paginating from.
     * @param perPage Number of monitors to return per page.
     * @param sort String for sort order, composed of field and sort order separate by a comma, e.g. &#x60;name,asc&#x60;. Supported sort directions: &#x60;asc&#x60;, &#x60;desc&#x60;. Supported fields:  * &#x60;name&#x60; * &#x60;status&#x60; * &#x60;tags&#x60;
     */
    PromiseMonitorsApi.prototype.searchMonitors = function (query, page, perPage, sort, options) {
        var result = this.api.searchMonitors(query, page, perPage, sort, options);
        return result.toPromise();
    };
    /**
     * Edit the specified monitor.
     * Edit a monitor
     * @param monitorId The ID of the monitor.
     * @param body Edit a monitor request body.
     */
    PromiseMonitorsApi.prototype.updateMonitor = function (monitorId, body, options) {
        var result = this.api.updateMonitor(monitorId, body, options);
        return result.toPromise();
    };
    /**
     * Validate the monitor provided in the request.
     * Validate a monitor
     * @param body Monitor request object
     */
    PromiseMonitorsApi.prototype.validateMonitor = function (body, options) {
        var result = this.api.validateMonitor(body, options);
        return result.toPromise();
    };
    return PromiseMonitorsApi;
}());
exports.PromiseMonitorsApi = PromiseMonitorsApi;
var ObservableAPI_18 = require("./ObservableAPI");
var PromiseNotebooksApi = /** @class */ (function () {
    function PromiseNotebooksApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_18.ObservableNotebooksApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Create a notebook using the specified options.
     * Create a notebook
     * @param body The JSON description of the notebook you want to create.
     */
    PromiseNotebooksApi.prototype.createNotebook = function (body, options) {
        var result = this.api.createNotebook(body, options);
        return result.toPromise();
    };
    /**
     * Delete a notebook using the specified ID.
     * Delete a notebook
     * @param notebookId Unique ID, assigned when you create the notebook.
     */
    PromiseNotebooksApi.prototype.deleteNotebook = function (notebookId, options) {
        var result = this.api.deleteNotebook(notebookId, options);
        return result.toPromise();
    };
    /**
     * Get a notebook using the specified notebook ID.
     * Get a notebook
     * @param notebookId Unique ID, assigned when you create the notebook.
     */
    PromiseNotebooksApi.prototype.getNotebook = function (notebookId, options) {
        var result = this.api.getNotebook(notebookId, options);
        return result.toPromise();
    };
    /**
     * Get all notebooks. This can also be used to search for notebooks with a particular `query` in the notebook `name` or author `handle`.
     * Get all notebooks
     * @param authorHandle Return notebooks created by the given &#x60;author_handle&#x60;.
     * @param excludeAuthorHandle Return notebooks not created by the given &#x60;author_handle&#x60;.
     * @param start The index of the first notebook you want returned.
     * @param count The number of notebooks to be returned.
     * @param sortField Sort by field &#x60;modified&#x60; or &#x60;name&#x60;.
     * @param sortDir Sort by direction &#x60;asc&#x60; or &#x60;desc&#x60;.
     * @param query Return only notebooks with &#x60;query&#x60; string in notebook name or author handle.
     * @param includeCells Value of &#x60;false&#x60; excludes the &#x60;cells&#x60; and global &#x60;time&#x60; for each notebook.
     */
    PromiseNotebooksApi.prototype.listNotebooks = function (authorHandle, excludeAuthorHandle, start, count, sortField, sortDir, query, includeCells, options) {
        var result = this.api.listNotebooks(authorHandle, excludeAuthorHandle, start, count, sortField, sortDir, query, includeCells, options);
        return result.toPromise();
    };
    /**
     * Update a notebook using the specified ID.
     * Update a notebook
     * @param notebookId Unique ID, assigned when you create the notebook.
     * @param body Update notebook request body.
     */
    PromiseNotebooksApi.prototype.updateNotebook = function (notebookId, body, options) {
        var result = this.api.updateNotebook(notebookId, body, options);
        return result.toPromise();
    };
    return PromiseNotebooksApi;
}());
exports.PromiseNotebooksApi = PromiseNotebooksApi;
var ObservableAPI_19 = require("./ObservableAPI");
var PromiseOrganizationsApi = /** @class */ (function () {
    function PromiseOrganizationsApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_19.ObservableOrganizationsApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Create a child organization.  This endpoint requires the [multi-organization account](https://docs.datadoghq.com/account_management/multi_organization/) feature and must be enabled by [contacting support](https://docs.datadoghq.com/help/).  Once a new child organization is created, you can interact with it by using the `org.public_id`, `pi_key.key`, and `application_key.hash` provided in the response.
     * Create a child organization
     * @param body Organization object that needs to be created
     */
    PromiseOrganizationsApi.prototype.createChildOrg = function (body, options) {
        var result = this.api.createChildOrg(body, options);
        return result.toPromise();
    };
    /**
     * Get organization information.
     * Get organization information
     * @param publicId The &#x60;public_id&#x60; of the organization you are operating within.
     */
    PromiseOrganizationsApi.prototype.getOrg = function (publicId, options) {
        var result = this.api.getOrg(publicId, options);
        return result.toPromise();
    };
    /**
     * List your managed organizations.
     * List your managed organizations
     */
    PromiseOrganizationsApi.prototype.listOrgs = function (options) {
        var result = this.api.listOrgs(options);
        return result.toPromise();
    };
    /**
     * Update your organization.
     * Update your organization
     * @param publicId The &#x60;public_id&#x60; of the organization you are operating within.
     * @param body
     */
    PromiseOrganizationsApi.prototype.updateOrg = function (publicId, body, options) {
        var result = this.api.updateOrg(publicId, body, options);
        return result.toPromise();
    };
    /**
     * There are a couple of options for updating the Identity Provider (IdP) metadata from your SAML IdP.  * **Multipart Form-Data**: Post the IdP metadata file using a form post.  * **XML Body:** Post the IdP metadata file as the body of the request.
     * Upload IdP metadata
     * @param publicId The &#x60;public_id&#x60; of the organization you are operating with
     * @param idpFile The path to the XML metadata file you wish to upload.
     */
    PromiseOrganizationsApi.prototype.uploadIdPForOrg = function (publicId, idpFile, options) {
        var result = this.api.uploadIdPForOrg(publicId, idpFile, options);
        return result.toPromise();
    };
    return PromiseOrganizationsApi;
}());
exports.PromiseOrganizationsApi = PromiseOrganizationsApi;
var ObservableAPI_20 = require("./ObservableAPI");
var PromisePagerDutyIntegrationApi = /** @class */ (function () {
    function PromisePagerDutyIntegrationApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_20.ObservablePagerDutyIntegrationApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Create a new service object in the PagerDuty integration.
     * Create a new service object
     * @param body Create a new service object request body.
     */
    PromisePagerDutyIntegrationApi.prototype.createPagerDutyIntegrationService = function (body, options) {
        var result = this.api.createPagerDutyIntegrationService(body, options);
        return result.toPromise();
    };
    /**
     * Delete a single service object in the Datadog-PagerDuty integration.
     * Delete a single service object
     * @param serviceName The service name
     */
    PromisePagerDutyIntegrationApi.prototype.deletePagerDutyIntegrationService = function (serviceName, options) {
        var result = this.api.deletePagerDutyIntegrationService(serviceName, options);
        return result.toPromise();
    };
    /**
     * Get service name in the Datadog-PagerDuty integration.
     * Get a single service object
     * @param serviceName The service name.
     */
    PromisePagerDutyIntegrationApi.prototype.getPagerDutyIntegrationService = function (serviceName, options) {
        var result = this.api.getPagerDutyIntegrationService(serviceName, options);
        return result.toPromise();
    };
    /**
     * Update a single service object in the Datadog-PagerDuty integration.
     * Update a single service object
     * @param serviceName The service name
     * @param body Update an existing service object request body.
     */
    PromisePagerDutyIntegrationApi.prototype.updatePagerDutyIntegrationService = function (serviceName, body, options) {
        var result = this.api.updatePagerDutyIntegrationService(serviceName, body, options);
        return result.toPromise();
    };
    return PromisePagerDutyIntegrationApi;
}());
exports.PromisePagerDutyIntegrationApi = PromisePagerDutyIntegrationApi;
var ObservableAPI_21 = require("./ObservableAPI");
var PromiseServiceChecksApi = /** @class */ (function () {
    function PromiseServiceChecksApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_21.ObservableServiceChecksApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Submit a list of Service Checks.  **Note**: A valid API key is required.
     * Submit a Service Check
     * @param body Service Check request body.
     */
    PromiseServiceChecksApi.prototype.submitServiceCheck = function (body, options) {
        var result = this.api.submitServiceCheck(body, options);
        return result.toPromise();
    };
    return PromiseServiceChecksApi;
}());
exports.PromiseServiceChecksApi = PromiseServiceChecksApi;
var ObservableAPI_22 = require("./ObservableAPI");
var PromiseServiceLevelObjectiveCorrectionsApi = /** @class */ (function () {
    function PromiseServiceLevelObjectiveCorrectionsApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_22.ObservableServiceLevelObjectiveCorrectionsApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Create an SLO Correction
     * Create an SLO correction
     * @param body Create an SLO Correction
     */
    PromiseServiceLevelObjectiveCorrectionsApi.prototype.createSLOCorrection = function (body, options) {
        var result = this.api.createSLOCorrection(body, options);
        return result.toPromise();
    };
    /**
     * Permanently delete the specified SLO correction object
     * Delete an SLO correction
     * @param sloCorrectionId The ID of the SLO correction object
     */
    PromiseServiceLevelObjectiveCorrectionsApi.prototype.deleteSLOCorrection = function (sloCorrectionId, options) {
        var result = this.api.deleteSLOCorrection(sloCorrectionId, options);
        return result.toPromise();
    };
    /**
     * Get an SLO correction
     * Get an SLO correction for an SLO
     * @param sloCorrectionId The ID of the SLO correction object
     */
    PromiseServiceLevelObjectiveCorrectionsApi.prototype.getSLOCorrection = function (sloCorrectionId, options) {
        var result = this.api.getSLOCorrection(sloCorrectionId, options);
        return result.toPromise();
    };
    /**
     * Get all Service Level Objective corrections
     * Get all SLO corrections
     */
    PromiseServiceLevelObjectiveCorrectionsApi.prototype.listSLOCorrection = function (options) {
        var result = this.api.listSLOCorrection(options);
        return result.toPromise();
    };
    /**
     * Update the specified SLO correction object object
     * Update an SLO correction
     * @param sloCorrectionId The ID of the SLO correction object
     * @param body The edited SLO correction object.
     */
    PromiseServiceLevelObjectiveCorrectionsApi.prototype.updateSLOCorrection = function (sloCorrectionId, body, options) {
        var result = this.api.updateSLOCorrection(sloCorrectionId, body, options);
        return result.toPromise();
    };
    return PromiseServiceLevelObjectiveCorrectionsApi;
}());
exports.PromiseServiceLevelObjectiveCorrectionsApi = PromiseServiceLevelObjectiveCorrectionsApi;
var ObservableAPI_23 = require("./ObservableAPI");
var PromiseServiceLevelObjectivesApi = /** @class */ (function () {
    function PromiseServiceLevelObjectivesApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_23.ObservableServiceLevelObjectivesApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Check if an SLO can be safely deleted. For example, assure an SLO can be deleted without disrupting a dashboard.
     * Check if SLOs can be safely deleted
     * @param ids A comma separated list of the IDs of the service level objectives objects.
     */
    PromiseServiceLevelObjectivesApi.prototype.checkCanDeleteSLO = function (ids, options) {
        var result = this.api.checkCanDeleteSLO(ids, options);
        return result.toPromise();
    };
    /**
     * Create a service level objective object.
     * Create an SLO object
     * @param body Service level objective request object.
     */
    PromiseServiceLevelObjectivesApi.prototype.createSLO = function (body, options) {
        var result = this.api.createSLO(body, options);
        return result.toPromise();
    };
    /**
     * Permanently delete the specified service level objective object.  If an SLO is used in a dashboard, the `DELETE /v1/slo/` endpoint returns a 409 conflict error because the SLO is referenced in a dashboard.
     * Delete an SLO
     * @param sloId The ID of the service level objective.
     * @param force Delete the monitor even if it&#39;s referenced by other resources (e.g. SLO, composite monitor).
     */
    PromiseServiceLevelObjectivesApi.prototype.deleteSLO = function (sloId, force, options) {
        var result = this.api.deleteSLO(sloId, force, options);
        return result.toPromise();
    };
    /**
     * Delete (or partially delete) multiple service level objective objects.  This endpoint facilitates deletion of one or more thresholds for one or more service level objective objects. If all thresholds are deleted, the service level objective object is deleted as well.
     * Bulk Delete SLO Timeframes
     * @param body Delete multiple service level objective objects request body.
     */
    PromiseServiceLevelObjectivesApi.prototype.deleteSLOTimeframeInBulk = function (body, options) {
        var result = this.api.deleteSLOTimeframeInBulk(body, options);
        return result.toPromise();
    };
    /**
     * Get a service level objective object.
     * Get an SLO's details
     * @param sloId The ID of the service level objective object.
     * @param withConfiguredAlertIds Get the IDs of SLO monitors that reference this SLO.
     */
    PromiseServiceLevelObjectivesApi.prototype.getSLO = function (sloId, withConfiguredAlertIds, options) {
        var result = this.api.getSLO(sloId, withConfiguredAlertIds, options);
        return result.toPromise();
    };
    /**
     * Get a specific SLO’s history, regardless of its SLO type.  The detailed history data is structured according to the source data type. For example, metric data is included for event SLOs that use the metric source, and monitor SLO types include the monitor transition history.  **Note:** There are different response formats for event based and time based SLOs. Examples of both are shown.
     * Get an SLO's history
     * @param sloId The ID of the service level objective object.
     * @param fromTs The &#x60;from&#x60; timestamp for the query window in epoch seconds.
     * @param toTs The &#x60;to&#x60; timestamp for the query window in epoch seconds.
     * @param target The SLO target. If &#x60;target&#x60; is passed in, the response will include the error budget that remains.
     */
    PromiseServiceLevelObjectivesApi.prototype.getSLOHistory = function (sloId, fromTs, toTs, target, options) {
        var result = this.api.getSLOHistory(sloId, fromTs, toTs, target, options);
        return result.toPromise();
    };
    /**
     * Get a list of service level objective objects for your organization.
     * Get all SLOs
     * @param ids A comma separated list of the IDs of the service level objectives objects.
     * @param query The query string to filter results based on SLO names.
     * @param tagsQuery The query string to filter results based on a single SLO tag.
     * @param metricsQuery The query string to filter results based on SLO numerator and denominator.
     */
    PromiseServiceLevelObjectivesApi.prototype.listSLOs = function (ids, query, tagsQuery, metricsQuery, options) {
        var result = this.api.listSLOs(ids, query, tagsQuery, metricsQuery, options);
        return result.toPromise();
    };
    /**
     * Update the specified service level objective object.
     * Update an SLO
     * @param sloId The ID of the service level objective object.
     * @param body The edited service level objective request object.
     */
    PromiseServiceLevelObjectivesApi.prototype.updateSLO = function (sloId, body, options) {
        var result = this.api.updateSLO(sloId, body, options);
        return result.toPromise();
    };
    return PromiseServiceLevelObjectivesApi;
}());
exports.PromiseServiceLevelObjectivesApi = PromiseServiceLevelObjectivesApi;
var ObservableAPI_24 = require("./ObservableAPI");
var PromiseSlackIntegrationApi = /** @class */ (function () {
    function PromiseSlackIntegrationApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_24.ObservableSlackIntegrationApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Add a channel to your Datadog-Slack integration.
     * Create a Slack integration channel
     * @param accountName Your Slack account name.
     * @param body Payload describing Slack channel to be created
     */
    PromiseSlackIntegrationApi.prototype.createSlackIntegrationChannel = function (accountName, body, options) {
        var result = this.api.createSlackIntegrationChannel(accountName, body, options);
        return result.toPromise();
    };
    /**
     * Get a channel configured for your Datadog-Slack integration.
     * Get a Slack integration channel
     * @param accountName Your Slack account name.
     * @param channelName The name of the Slack channel being operated on.
     */
    PromiseSlackIntegrationApi.prototype.getSlackIntegrationChannel = function (accountName, channelName, options) {
        var result = this.api.getSlackIntegrationChannel(accountName, channelName, options);
        return result.toPromise();
    };
    /**
     * Get a list of all channels configured for your Datadog-Slack integration.
     * Get all channels in a Slack integration
     * @param accountName Your Slack account name.
     */
    PromiseSlackIntegrationApi.prototype.getSlackIntegrationChannels = function (accountName, options) {
        var result = this.api.getSlackIntegrationChannels(accountName, options);
        return result.toPromise();
    };
    /**
     * Remove a channel from your Datadog-Slack integration.
     * Remove a Slack integration channel
     * @param accountName Your Slack account name.
     * @param channelName The name of the Slack channel being operated on.
     */
    PromiseSlackIntegrationApi.prototype.removeSlackIntegrationChannel = function (accountName, channelName, options) {
        var result = this.api.removeSlackIntegrationChannel(accountName, channelName, options);
        return result.toPromise();
    };
    /**
     * Update a channel used in your Datadog-Slack integration.
     * Update a Slack integration channel
     * @param accountName Your Slack account name.
     * @param channelName The name of the Slack channel being operated on.
     * @param body Payload describing fields and values to be updated.
     */
    PromiseSlackIntegrationApi.prototype.updateSlackIntegrationChannel = function (accountName, channelName, body, options) {
        var result = this.api.updateSlackIntegrationChannel(accountName, channelName, body, options);
        return result.toPromise();
    };
    return PromiseSlackIntegrationApi;
}());
exports.PromiseSlackIntegrationApi = PromiseSlackIntegrationApi;
var ObservableAPI_25 = require("./ObservableAPI");
var PromiseSnapshotsApi = /** @class */ (function () {
    function PromiseSnapshotsApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_25.ObservableSnapshotsApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Take graph snapshots. **Note**: When a snapshot is created, there is some delay before it is available.
     * Take graph snapshots
     * @param start The POSIX timestamp of the start of the query.
     * @param end The POSIX timestamp of the end of the query.
     * @param metricQuery The metric query.
     * @param eventQuery A query that adds event bands to the graph.
     * @param graphDef A JSON document defining the graph. &#x60;graph_def&#x60; can be used instead of &#x60;metric_query&#x60;. The JSON document uses the [grammar defined here](https://docs.datadoghq.com/graphing/graphing_json/#grammar) and should be formatted to a single line then URL encoded.
     * @param title A title for the graph. If no title is specified, the graph does not have a title.
     */
    PromiseSnapshotsApi.prototype.getGraphSnapshot = function (start, end, metricQuery, eventQuery, graphDef, title, options) {
        var result = this.api.getGraphSnapshot(start, end, metricQuery, eventQuery, graphDef, title, options);
        return result.toPromise();
    };
    return PromiseSnapshotsApi;
}());
exports.PromiseSnapshotsApi = PromiseSnapshotsApi;
var ObservableAPI_26 = require("./ObservableAPI");
var PromiseSyntheticsApi = /** @class */ (function () {
    function PromiseSyntheticsApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_26.ObservableSyntheticsApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Create a Synthetics global variable.
     * Create a global variable
     * @param body Details of the global variable to create.
     */
    PromiseSyntheticsApi.prototype.createGlobalVariable = function (body, options) {
        var result = this.api.createGlobalVariable(body, options);
        return result.toPromise();
    };
    /**
     * Create a new Synthetics private location.
     * Create a private location
     * @param body Details of the private location to create.
     */
    PromiseSyntheticsApi.prototype.createPrivateLocation = function (body, options) {
        var result = this.api.createPrivateLocation(body, options);
        return result.toPromise();
    };
    /**
     * Create a Synthetic API test.
     * Create an API test
     * @param body Details of the test to create.
     */
    PromiseSyntheticsApi.prototype.createSyntheticsAPITest = function (body, options) {
        var result = this.api.createSyntheticsAPITest(body, options);
        return result.toPromise();
    };
    /**
     * Create a Synthetic browser test.
     * Create a browser test
     * @param body Details of the test to create.
     */
    PromiseSyntheticsApi.prototype.createSyntheticsBrowserTest = function (body, options) {
        var result = this.api.createSyntheticsBrowserTest(body, options);
        return result.toPromise();
    };
    /**
     * Delete a Synthetics global variable.
     * Delete a global variable
     * @param variableId The ID of the global variable.
     */
    PromiseSyntheticsApi.prototype.deleteGlobalVariable = function (variableId, options) {
        var result = this.api.deleteGlobalVariable(variableId, options);
        return result.toPromise();
    };
    /**
     * Delete a Synthetics private location.
     * Delete a private location
     * @param locationId The ID of the private location.
     */
    PromiseSyntheticsApi.prototype.deletePrivateLocation = function (locationId, options) {
        var result = this.api.deletePrivateLocation(locationId, options);
        return result.toPromise();
    };
    /**
     * Delete multiple Synthetic tests by ID.
     * Delete tests
     * @param body Public ID list of the Synthetic tests to be deleted.
     */
    PromiseSyntheticsApi.prototype.deleteTests = function (body, options) {
        var result = this.api.deleteTests(body, options);
        return result.toPromise();
    };
    /**
     * Edit a Synthetics global variable.
     * Edit a global variable
     * @param variableId The ID of the global variable.
     * @param body Details of the global variable to update.
     */
    PromiseSyntheticsApi.prototype.editGlobalVariable = function (variableId, body, options) {
        var result = this.api.editGlobalVariable(variableId, body, options);
        return result.toPromise();
    };
    /**
     * Get the detailed configuration associated with a Synthetic API test.
     * Get an API test
     * @param publicId The public ID of the test to get details from.
     */
    PromiseSyntheticsApi.prototype.getAPITest = function (publicId, options) {
        var result = this.api.getAPITest(publicId, options);
        return result.toPromise();
    };
    /**
     * Get the last 50 test results summaries for a given Synthetics API test.
     * Get an API test's latest results summaries
     * @param publicId The public ID of the test for which to search results for.
     * @param fromTs Timestamp from which to start querying results.
     * @param toTs Timestamp up to which to query results.
     * @param probeDc Locations for which to query results.
     */
    PromiseSyntheticsApi.prototype.getAPITestLatestResults = function (publicId, fromTs, toTs, probeDc, options) {
        var result = this.api.getAPITestLatestResults(publicId, fromTs, toTs, probeDc, options);
        return result.toPromise();
    };
    /**
     * Get a specific full result from a given (API) Synthetic test.
     * Get an API test result
     * @param publicId The public ID of the API test to which the target result belongs.
     * @param resultId The ID of the result to get.
     */
    PromiseSyntheticsApi.prototype.getAPITestResult = function (publicId, resultId, options) {
        var result = this.api.getAPITestResult(publicId, resultId, options);
        return result.toPromise();
    };
    /**
     * Get the detailed configuration (including steps) associated with a Synthetic browser test.
     * Get a browser test
     * @param publicId The public ID of the test to get details from.
     */
    PromiseSyntheticsApi.prototype.getBrowserTest = function (publicId, options) {
        var result = this.api.getBrowserTest(publicId, options);
        return result.toPromise();
    };
    /**
     * Get the last 50 test results summaries for a given Synthetics Browser test.
     * Get a browser test's latest results summaries
     * @param publicId The public ID of the browser test for which to search results for.
     * @param fromTs Timestamp from which to start querying results.
     * @param toTs Timestamp up to which to query results.
     * @param probeDc Locations for which to query results.
     */
    PromiseSyntheticsApi.prototype.getBrowserTestLatestResults = function (publicId, fromTs, toTs, probeDc, options) {
        var result = this.api.getBrowserTestLatestResults(publicId, fromTs, toTs, probeDc, options);
        return result.toPromise();
    };
    /**
     * Get a specific full result from a given (browser) Synthetic test.
     * Get a browser test result
     * @param publicId The public ID of the browser test to which the target result belongs.
     * @param resultId The ID of the result to get.
     */
    PromiseSyntheticsApi.prototype.getBrowserTestResult = function (publicId, resultId, options) {
        var result = this.api.getBrowserTestResult(publicId, resultId, options);
        return result.toPromise();
    };
    /**
     * Get the detailed configuration of a global variable.
     * Get a global variable
     * @param variableId The ID of the global variable.
     */
    PromiseSyntheticsApi.prototype.getGlobalVariable = function (variableId, options) {
        var result = this.api.getGlobalVariable(variableId, options);
        return result.toPromise();
    };
    /**
     * Get a Synthetics private location.
     * Get a private location
     * @param locationId The ID of the private location.
     */
    PromiseSyntheticsApi.prototype.getPrivateLocation = function (locationId, options) {
        var result = this.api.getPrivateLocation(locationId, options);
        return result.toPromise();
    };
    /**
     * Get the detailed configuration associated with a Synthetics test.
     * Get a test configuration
     * @param publicId The public ID of the test to get details from.
     */
    PromiseSyntheticsApi.prototype.getTest = function (publicId, options) {
        var result = this.api.getTest(publicId, options);
        return result.toPromise();
    };
    /**
     * Get the list of all Synthetics global variables.
     * Get all global variables
     */
    PromiseSyntheticsApi.prototype.listGlobalVariables = function (options) {
        var result = this.api.listGlobalVariables(options);
        return result.toPromise();
    };
    /**
     * Get the list of public and private locations available for Synthetic tests. No arguments required.
     * Get all locations (public and private)
     */
    PromiseSyntheticsApi.prototype.listLocations = function (options) {
        var result = this.api.listLocations(options);
        return result.toPromise();
    };
    /**
     * Get the list of all Synthetic tests.
     * Get the list of all tests
     */
    PromiseSyntheticsApi.prototype.listTests = function (options) {
        var result = this.api.listTests(options);
        return result.toPromise();
    };
    /**
     * Trigger a set of Synthetics tests for continuous integration.
     * Trigger tests from CI/CD pipelines
     * @param body Details of the test to trigger.
     */
    PromiseSyntheticsApi.prototype.triggerCITests = function (body, options) {
        var result = this.api.triggerCITests(body, options);
        return result.toPromise();
    };
    /**
     * Edit the configuration of a Synthetic API test.
     * Edit an API test
     * @param publicId The public ID of the test to get details from.
     * @param body New test details to be saved.
     */
    PromiseSyntheticsApi.prototype.updateAPITest = function (publicId, body, options) {
        var result = this.api.updateAPITest(publicId, body, options);
        return result.toPromise();
    };
    /**
     * Edit the configuration of a Synthetic browser test.
     * Edit a browser test
     * @param publicId The public ID of the test to get details from.
     * @param body New test details to be saved.
     */
    PromiseSyntheticsApi.prototype.updateBrowserTest = function (publicId, body, options) {
        var result = this.api.updateBrowserTest(publicId, body, options);
        return result.toPromise();
    };
    /**
     * Edit a Synthetics private location.
     * Edit a private location
     * @param locationId The ID of the private location.
     * @param body Details of the private location to be updated.
     */
    PromiseSyntheticsApi.prototype.updatePrivateLocation = function (locationId, body, options) {
        var result = this.api.updatePrivateLocation(locationId, body, options);
        return result.toPromise();
    };
    /**
     * Pause or start a Synthetics test by changing the status.
     * Pause or start a test
     * @param publicId The public ID of the Synthetic test to update.
     * @param body Status to set the given Synthetic test to.
     */
    PromiseSyntheticsApi.prototype.updateTestPauseStatus = function (publicId, body, options) {
        var result = this.api.updateTestPauseStatus(publicId, body, options);
        return result.toPromise();
    };
    return PromiseSyntheticsApi;
}());
exports.PromiseSyntheticsApi = PromiseSyntheticsApi;
var ObservableAPI_27 = require("./ObservableAPI");
var PromiseTagsApi = /** @class */ (function () {
    function PromiseTagsApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_27.ObservableTagsApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * This endpoint allows you to add new tags to a host, optionally specifying where these tags come from.
     * Add tags to a host
     * @param hostName This endpoint allows you to add new tags to a host, optionally specifying where the tags came from.
     * @param body Update host tags request body.
     * @param source The source of the tags. [Complete list of source attribute values](https://docs.datadoghq.com/integrations/faq/list-of-api-source-attribute-value).
     */
    PromiseTagsApi.prototype.createHostTags = function (hostName, body, source, options) {
        var result = this.api.createHostTags(hostName, body, source, options);
        return result.toPromise();
    };
    /**
     * This endpoint allows you to remove all user-assigned tags for a single host.
     * Remove host tags
     * @param hostName This endpoint allows you to remove all user-assigned tags for a single host.
     * @param source The source of the tags (e.g. chef, puppet). [Complete list of source attribute values](https://docs.datadoghq.com/integrations/faq/list-of-api-source-attribute-value).
     */
    PromiseTagsApi.prototype.deleteHostTags = function (hostName, source, options) {
        var result = this.api.deleteHostTags(hostName, source, options);
        return result.toPromise();
    };
    /**
     * Return the list of tags that apply to a given host.
     * Get host tags
     * @param hostName When specified, filters list of tags to those tags with the specified source.
     * @param source Source to filter.
     */
    PromiseTagsApi.prototype.getHostTags = function (hostName, source, options) {
        var result = this.api.getHostTags(hostName, source, options);
        return result.toPromise();
    };
    /**
     * Return a mapping of tags to hosts for your whole infrastructure.
     * Get Tags
     * @param source When specified, filters host list to those tags with the specified source.
     */
    PromiseTagsApi.prototype.listHostTags = function (source, options) {
        var result = this.api.listHostTags(source, options);
        return result.toPromise();
    };
    /**
     * This endpoint allows you to update/replace all tags in an integration source with those supplied in the request.
     * Update host tags
     * @param hostName This endpoint allows you to update/replace all in an integration source with those supplied in the request.
     * @param body Add tags to host
     * @param source The source of the tags (e.g. chef, puppet). [Complete list of source attribute values](https://docs.datadoghq.com/integrations/faq/list-of-api-source-attribute-value)
     */
    PromiseTagsApi.prototype.updateHostTags = function (hostName, body, source, options) {
        var result = this.api.updateHostTags(hostName, body, source, options);
        return result.toPromise();
    };
    return PromiseTagsApi;
}());
exports.PromiseTagsApi = PromiseTagsApi;
var ObservableAPI_28 = require("./ObservableAPI");
var PromiseUsageMeteringApi = /** @class */ (function () {
    function PromiseUsageMeteringApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_28.ObservableUsageMeteringApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Get daily custom reports.
     * Get the list of available daily custom reports
     * @param pageSize The number of files to return in the response. &#x60;[default&#x3D;60]&#x60;.
     * @param pageNumber The identifier of the first page to return. This parameter is used for the pagination feature &#x60;[default&#x3D;0]&#x60;.
     * @param sortDir The direction to sort by: &#x60;[desc, asc]&#x60;.
     * @param sort The field to sort by: &#x60;[computed_on, size, start_date, end_date]&#x60;.
     */
    PromiseUsageMeteringApi.prototype.getDailyCustomReports = function (pageSize, pageNumber, sortDir, sort, options) {
        var result = this.api.getDailyCustomReports(pageSize, pageNumber, sortDir, sort, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for incident management.
     * Get hourly usage for incident management
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getIncidentManagement = function (startHr, endHr, options) {
        var result = this.api.getIncidentManagement(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for ingested spans.
     * Get hourly usage for ingested spans
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getIngestedSpans = function (startHr, endHr, options) {
        var result = this.api.getIngestedSpans(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get monthly custom reports.
     * Get the list of available monthly custom reports
     * @param pageSize The number of files to return in the response &#x60;[default&#x3D;60].&#x60;
     * @param pageNumber The identifier of the first page to return. This parameter is used for the pagination feature &#x60;[default&#x3D;0]&#x60;.
     * @param sortDir The direction to sort by: &#x60;[desc, asc]&#x60;.
     * @param sort The field to sort by: &#x60;[computed_on, size, start_date, end_date]&#x60;.
     */
    PromiseUsageMeteringApi.prototype.getMonthlyCustomReports = function (pageSize, pageNumber, sortDir, sort, options) {
        var result = this.api.getMonthlyCustomReports(pageSize, pageNumber, sortDir, sort, options);
        return result.toPromise();
    };
    /**
     * Get specified daily custom reports.
     * Get specified daily custom reports
     * @param reportId The specified ID to search results for.
     */
    PromiseUsageMeteringApi.prototype.getSpecifiedDailyCustomReports = function (reportId, options) {
        var result = this.api.getSpecifiedDailyCustomReports(reportId, options);
        return result.toPromise();
    };
    /**
     * Get specified monthly custom reports.
     * Get specified monthly custom reports
     * @param reportId The specified ID to search results for.
     */
    PromiseUsageMeteringApi.prototype.getSpecifiedMonthlyCustomReports = function (reportId, options) {
        var result = this.api.getSpecifiedMonthlyCustomReports(reportId, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for tracing without limits.  **Note** This endpoint has been renamed to `/api/v1/usage/ingested-spans`.
     * Get hourly usage for tracing without limits
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getTracingWithoutLimits = function (startHr, endHr, options) {
        var result = this.api.getTracingWithoutLimits(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for analyzed logs (Security Monitoring).
     * Get hourly usage for analyzed logs
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageAnalyzedLogs = function (startHr, endHr, options) {
        var result = this.api.getUsageAnalyzedLogs(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get Usage Attribution.
     * Get Usage Attribution
     * @param startMonth Datetime in ISO-8601 format, UTC, precise to month: &#x60;[YYYY-MM]&#x60; for usage beginning in this month. Maximum of 15 months ago.
     * @param fields Comma-separated list of usage types to return, or &#x60;*&#x60; for all usage types.
     * @param endMonth Datetime in ISO-8601 format, UTC, precise to month: &#x60;[YYYY-MM]&#x60; for usage ending this month.
     * @param sortDirection The direction to sort by: &#x60;[desc, asc]&#x60;.
     * @param sortName The field to sort by.
     */
    PromiseUsageMeteringApi.prototype.getUsageAttribution = function (startMonth, fields, endMonth, sortDirection, sortName, options) {
        var result = this.api.getUsageAttribution(startMonth, fields, endMonth, sortDirection, sortName, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for audit logs.
     * Get hourly usage for audit logs
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageAuditLogs = function (startHr, endHr, options) {
        var result = this.api.getUsageAuditLogs(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get billable usage across your account.
     * Get billable usage across your account
     * @param month Datetime in ISO-8601 format, UTC, precise to month: &#x60;[YYYY-MM]&#x60; for usage starting this month.
     */
    PromiseUsageMeteringApi.prototype.getUsageBillableSummary = function (month, options) {
        var result = this.api.getUsageBillableSummary(month, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for Cloud Workload Security.
     * Get hourly usage for Cloud Workload Security
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageCWS = function (startHr, endHr, options) {
        var result = this.api.getUsageCWS(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for Cloud Security Posture Management (CSPM).
     * Get hourly usage for CSPM
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageCloudSecurityPostureManagement = function (startHr, endHr, options) {
        var result = this.api.getUsageCloudSecurityPostureManagement(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for [Fargate](https://docs.datadoghq.com/integrations/ecs_fargate/).
     * Get hourly usage for Fargate
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageFargate = function (startHr, endHr, options) {
        var result = this.api.getUsageFargate(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for hosts and containers.
     * Get hourly usage for hosts and containers
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageHosts = function (startHr, endHr, options) {
        var result = this.api.getUsageHosts(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for indexed spans.
     * Get hourly usage for indexed spans
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageIndexedSpans = function (startHr, endHr, options) {
        var result = this.api.getUsageIndexedSpans(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for IoT.
     * Get hourly usage for IoT
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageInternetOfThings = function (startHr, endHr, options) {
        var result = this.api.getUsageInternetOfThings(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for lambda.
     * Get hourly usage for Lambda
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageLambda = function (startHr, endHr, options) {
        var result = this.api.getUsageLambda(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for logs.
     * Get hourly usage for Logs
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageLogs = function (startHr, endHr, options) {
        var result = this.api.getUsageLogs(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for logs by index.
     * Get hourly usage for Logs by Index
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage ending **before** this hour.
     * @param indexName Comma-separated list of log index names.
     */
    PromiseUsageMeteringApi.prototype.getUsageLogsByIndex = function (startHr, endHr, indexName, options) {
        var result = this.api.getUsageLogsByIndex(startHr, endHr, indexName, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for indexed logs by retention period.
     * Get hourly logs usage by retention
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageLogsByRetention = function (startHr, endHr, options) {
        var result = this.api.getUsageLogsByRetention(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for network flows.
     * Get hourly usage for Network Flows
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageNetworkFlows = function (startHr, endHr, options) {
        var result = this.api.getUsageNetworkFlows(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for network hosts.
     * Get hourly usage for Network Hosts
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageNetworkHosts = function (startHr, endHr, options) {
        var result = this.api.getUsageNetworkHosts(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for profiled hosts.
     * Get hourly usage for profiled hosts
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageProfiling = function (startHr, endHr, options) {
        var result = this.api.getUsageProfiling(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for [RUM](https://docs.datadoghq.com/real_user_monitoring/) Sessions.
     * Get hourly usage for RUM Sessions
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage ending **before** this hour.
     * @param type RUM type: &#x60;[browser, mobile]&#x60;. Defaults to &#x60;browser&#x60;.
     */
    PromiseUsageMeteringApi.prototype.getUsageRumSessions = function (startHr, endHr, type, options) {
        var result = this.api.getUsageRumSessions(startHr, endHr, type, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for SNMP devices.
     * Get hourly usage for SNMP devices
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: &#x60;[YYYY-MM-DDThh]&#x60; for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageSNMP = function (startHr, endHr, options) {
        var result = this.api.getUsageSNMP(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get usage across your multi-org account. You must have the multi-org feature enabled.
     * Get usage across your multi-org account
     * @param startMonth Datetime in ISO-8601 format, UTC, precise to month: &#x60;[YYYY-MM]&#x60; for usage beginning in this month. Maximum of 15 months ago.
     * @param endMonth Datetime in ISO-8601 format, UTC, precise to month: &#x60;[YYYY-MM]&#x60; for usage ending this month.
     * @param includeOrgDetails Include usage summaries for each sub-org.
     */
    PromiseUsageMeteringApi.prototype.getUsageSummary = function (startMonth, endMonth, includeOrgDetails, options) {
        var result = this.api.getUsageSummary(startMonth, endMonth, includeOrgDetails, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for [Synthetics checks](https://docs.datadoghq.com/synthetics/).
     * Get hourly usage for Synthetics Checks
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageSynthetics = function (startHr, endHr, options) {
        var result = this.api.getUsageSynthetics(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for [synthetics API checks](https://docs.datadoghq.com/synthetics/).
     * Get hourly usage for Synthetics API Checks
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageSyntheticsAPI = function (startHr, endHr, options) {
        var result = this.api.getUsageSyntheticsAPI(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for synthetics browser checks.
     * Get hourly usage for Synthetics Browser Checks
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageSyntheticsBrowser = function (startHr, endHr, options) {
        var result = this.api.getUsageSyntheticsBrowser(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for [custom metrics](https://docs.datadoghq.com/developers/metrics/custom_metrics/).
     * Get hourly usage for custom metrics
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageTimeseries = function (startHr, endHr, options) {
        var result = this.api.getUsageTimeseries(startHr, endHr, options);
        return result.toPromise();
    };
    /**
     * Get all [custom metrics](https://docs.datadoghq.com/developers/metrics/custom_metrics/) by hourly average. Use the month parameter to get a month-to-date data resolution or use the day parameter to get a daily resolution. One of the two is required, and only one of the two is allowed.
     * Get all custom metrics by hourly average
     * @param month Datetime in ISO-8601 format, UTC, precise to month: [YYYY-MM] for usage beginning at this hour. (Either month or day should be specified, but not both)
     * @param day Datetime in ISO-8601 format, UTC, precise to day: [YYYY-MM-DD] for usage beginning at this hour. (Either month or day should be specified, but not both)
     * @param names Comma-separated list of metric names.
     * @param limit Maximum number of results to return (between 1 and 5000) - defaults to 500 results if limit not specified.
     * @param nextRecordId List following results with a next_record_id provided in the previous query.
     */
    PromiseUsageMeteringApi.prototype.getUsageTopAvgMetrics = function (month, day, names, limit, nextRecordId, options) {
        var result = this.api.getUsageTopAvgMetrics(month, day, names, limit, nextRecordId, options);
        return result.toPromise();
    };
    /**
     * Get hourly usage for trace search.  **Note** This endpoint has been renamed to `/api/v1/usage/indexed-spans`.
     * Get hourly usage for Trace Search
     * @param startHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage beginning at this hour.
     * @param endHr Datetime in ISO-8601 format, UTC, precise to hour: [YYYY-MM-DDThh] for usage ending **before** this hour.
     */
    PromiseUsageMeteringApi.prototype.getUsageTrace = function (startHr, endHr, options) {
        var result = this.api.getUsageTrace(startHr, endHr, options);
        return result.toPromise();
    };
    return PromiseUsageMeteringApi;
}());
exports.PromiseUsageMeteringApi = PromiseUsageMeteringApi;
var ObservableAPI_29 = require("./ObservableAPI");
var PromiseUsersApi = /** @class */ (function () {
    function PromiseUsersApi(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_29.ObservableUsersApi(configuration, requestFactory, responseProcessor);
    }
    /**
     * Create a user for your organization.  **Note**: Users can only be created with the admin access role if application keys belong to administrators.
     * Create a user
     * @param body User object that needs to be created.
     */
    PromiseUsersApi.prototype.createUser = function (body, options) {
        var result = this.api.createUser(body, options);
        return result.toPromise();
    };
    /**
     * Delete a user from an organization.  **Note**: This endpoint can only be used with application keys belonging to administrators.
     * Disable a user
     * @param userHandle The handle of the user.
     */
    PromiseUsersApi.prototype.disableUser = function (userHandle, options) {
        var result = this.api.disableUser(userHandle, options);
        return result.toPromise();
    };
    /**
     * Get a user's details.
     * Get user details
     * @param userHandle The ID of the user.
     */
    PromiseUsersApi.prototype.getUser = function (userHandle, options) {
        var result = this.api.getUser(userHandle, options);
        return result.toPromise();
    };
    /**
     * List all users for your organization.
     * List all users
     */
    PromiseUsersApi.prototype.listUsers = function (options) {
        var result = this.api.listUsers(options);
        return result.toPromise();
    };
    /**
     * Update a user information.  **Note**: It can only be used with application keys belonging to administrators.
     * Update a user
     * @param userHandle The ID of the user.
     * @param body Description of the update.
     */
    PromiseUsersApi.prototype.updateUser = function (userHandle, body, options) {
        var result = this.api.updateUser(userHandle, body, options);
        return result.toPromise();
    };
    return PromiseUsersApi;
}());
exports.PromiseUsersApi = PromiseUsersApi;
